solution/index.js

```diff
@@ -0,0 +1,758 @@
+                                elData.task = liTask.firstChild.textContent; //addin updated task to data
+                                setDataToLocalStorage(); //update localstorage as well
+                        }
+            }  
+       }
+}
+
+
+
+function localStorageTaskRemover(liTask) //get a task and removes it from the localstorage **and the data object!** && from UserContentInputArray
+{
+
+    //remove to do tasks from data collectors
+    if(liTask.parentElement.id === "todo-tasks"){
+        //delete todo task from data object
+        data["todo"].forEach( (taskObj, i) => taskObj.task === liTask.firstChild.textContent
+                                                                    && taskObj.date === liTask.childNodes[1].childNodes[1].textContent 
+                                                                    && taskObj.time === liTask.childNodes[1].childNodes[2].textContent
+                                                                    ?  data["todo"].splice(i,1)
+                                                                    :  console.log(taskObj) ); 
+    }
+    
+
+    //remove in-progress tasks from data collectors
+    if(liTask.parentElement.id === "in-progress-tasks") {
+        //delete in-progress task from data object
+        data["inProgress"].forEach( (taskObj, i) => taskObj.task === liTask.firstChild.textContent 
+                                                                    && taskObj.date === liTask.childNodes[1].childNodes[1].textContent
+                                                                    && taskObj.time === liTask.childNodes[1].childNodes[2].textContent
+                                                                    ?  data["inProgress"].splice(i,1) 
+                                                                    :  console.log(taskObj));
+    }
+    
+
+    //remove done tasks from data collectors
+    if(liTask.parentElement.id === "done-tasks"){
+        //delete done task from data object
+        data["done"].forEach( (taskObj, i) => taskObj.task === liTask.firstChild.textContent
+                                                                    && taskObj.date === liTask.childNodes[1].childNodes[1].textContent 
+                                                                    && taskObj.time === liTask.childNodes[1].childNodes[2].textContent 
+                                                                    ?  data["done"].splice(i,1)
+                                                                    :  console.log(taskObj) );
+    }
+
+     //remove from localstorage todo === add updated data boject
+     setDataToLocalStorage();
+    
+
+ }
+    
+
+    
+
+    
+
+
+
+    // listItems.forEach((list) => { //observing the changes for every list
+    // observer.observe( list, {childList: true }); // observing for list items which being added
+    // });
+/** ~~ !!! STOP THE MUTANT WHEN USER NOT MAKING AN EVENT RELATED TO IT!!! ~~ */
```

Comment:
Those mutants are really getting out of hand.... We should bring the army to deal with them!